zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -176,24 +232,28 @@ object Cookie {
     val cookies: Array[String]  = headerValue.split(";").map(_.trim)
     val x: List[Option[Cookie]] = cookies.toList.map(a => {
       val (name, content) = splitNameContent(a)
-      if (name.trim == "" && content.isEmpty) None
-      else Some(Cookie(name, content.getOrElse("")))
+      if (name == "" && content.isEmpty) None
+      else Some(Cookie(name, content))
     })
 
     if (x.contains(None))
       None
     else Some(x.map(_.get))
   }
 
-  private def parseDate(v: String): Option[Instant] =
-    Try(Instant.parse(v)).toOption
-
-  private def splitNameContent(kv: String): (String, Option[String]) =
-    kv.split("=", 2).map(_.trim) match {
-      case Array(v1)     => (v1, None)
-      case Array(v1, v2) => (v1, Some(v2))
-      case _             => ("", None)
+  @inline
+  private def parseDate(v: String): Try[Instant] =
+    Try(Instant.parse(v))
+
+  @inline
+  private def splitNameContent(str: String): (String, String) = {
+    val i = str.indexOf('=')
+    if (i >= 0) {
+      (str.substring(0, i).trim, str.substring(i + 1).trim)
```

Comment:
Extremely expensive computations. Every `substring` and `trim` has a `O(n)` space cmplx. If you can reduce this it will make a big difference.

Reply:
Here is a crazy idea :P 

What if you keep the original string as it is and at the time of parsing create a cookie that maintains indexes of start and stop on the original string. Thereby not allocating any intermediary memory at all. eg:
```scala
trait Cookie {
  private def raw: String
  private def nameStart: Int = ??? // fill this at the time of parsing
  private def nameEnd: Int   = ??? // fill this at the time of parsing
  def name: String = raw.substring(nameStart, nameEnd)
}
```
The code above is just to depict the idea, actual implementation might be very different. However, this would be dramatically faster than the original solution because you are only allocating when you need that data.

If you would like to do this index-based optimization in a different PR, I wouldn't mind that also :)